### PR TITLE
V158 sidebox select form submit

### PR DESCRIPTION
**Pull request by lat9**
Partial (sidebox-only) corrections for #4879.

### CODE REVIEW

**Review by dbltoe — COMMENTED** [review]
Getting a partial blank page on home load after the script loading .categoryListBoxContents.

Will recheck all 12 files

**Comment by dbltoe** [comment]
My bad put the new file from responsive_classic and copied it into template_default instead of the actual one for template_default.

Only problem now is the disabled for the Please Select options and the Go buttons triggers ADA.

Could you use readonly versus disabled?

Options at https://myzencarthost.com/accessibility/andi/help/alerts.html?disabled_elements

**Comment by lat9** [comment]
I'll see regarding `readonly` vs `disabled`.  Since this is now controlled via jQuery, it might be OK.

**Comment by lat9** [comment]
> I'll see regarding `readonly` vs `disabled`. Since this is now controlled via jQuery, it might be OK.

Setting the attribute to `readonly` results in the browser allowing that option to be selected; tried both Firefox and Chrome on Windows.  That said, keeping the `disabled` attribute.

**Comment by dbltoe** [comment]
Perhaps an Arian tag or title announcing not an option?

**Comment by dbltoe** [comment]
From ADA suggestions:
`Instead of using the disabled attribute, keep the element in the tab order, use aria-disabled="true", prevent the element from functioning using JavaScript, and modify CSS to make the element look disabled (grayed out).`
We already have the grayed out part covered except for manufacturers.
I'll try to start over with my tests and try the aria-disabled.

**Comment by lat9** [comment]
> From ADA suggestions: `Instead of using the disabled attribute, keep the element in the tab order, use aria-disabled="true", prevent the element from functioning using JavaScript, and modify CSS to make the element look disabled (grayed out).` We already have the grayed out part covered except for manufacturers. I'll try to start over with my tests and try the aria-disabled.

I'll see what I can do in the jQuery script.

**Comment by lat9** [comment]
OK that change just pushed now takes an empty `''` option value and converts its text into an `<optgroup>` within which all other `<option>`'s are copied.  That makes the "Please select..." (er) unselectable.

**Comment by scottcwilson** [comment]
I'd love to see an article in the /about_us/ section of the FAQ on this
esp. with comparisons. 

On June 30, 2022, GitHub ***@***.***> wrote:
> I'm sorry this has been such a headache and time consumer for you.
>
> Thanks to your efforts, we are way closer to all 100s in Lighthouse
> AND ADA/WCAG compliance out of the box.
>
> I've tested almost all of the other popular carts and none come even
> close.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/zencart/zencart/pull/4932#issuecomment-1171541916>,
or unsubscribe <https://github.com/notifications/unsubscribe-
auth/ABBQFVXKI4FHIDESRUOO7PTVRXRKRANCNFSM5Z6LLMMA>.
You are receiving this because you commented.Message ID:
***@***.***>

**Comment by lat9** [comment]
@dbltoe, does your previous comment indicate that these changes are ready for merge?

**Comment by dbltoe** [comment]
Will test shortly.

**Comment by dbltoe** [comment]
I'm not finding a comment I thought I posted.

All is working great.

I would add two lines to the stylesheet_colors.css or stylesheet.css (both lines are color but are in line with what this PR adds to the stylesheet.css.

`optgroup[label="Please Select"] {color: #DB0000;}
option{color: #000000;}`

This gives a not-too-glaring red to the Please Select option.  Might be a good hint to the user.

File: includes/modules/sideboxes/manufacturers.php [hunk 1]
```diff
@@ -8,58 +8,53 @@
...
+    $manufacturer_sidebox_query =
+        "SELECT m.manufacturers_id, m.manufacturers_name
+           FROM " . TABLE_MANUFACTURERS . " m
+           ORDER BY manufacturers_name";
+}
+
+$manufacturer_sidebox = $db->Execute($manufacturer_sidebox_query);
+
+if (!$manufacturer_sidebox->EOF) {
+    $number_of_rows = $manufacturer_sidebox->RecordCount() + 1;
```
[the remark this answers is not in the document]
Comment (lat9) [reply includes/modules/sideboxes/manufacturers.php]: That update just pushed removes those unused elements.

**Pull request merged**